core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -45,25 +49,31 @@ public String getType() {
         return type;
     }
 
+    public String getMatch() {
+        return match;
+    }
+
     public Object getValue() {
         return value;
     }
 
-    public boolean evaluate(Map<String, ?> attributes) {
+    public @Nullable Boolean evaluate(Map<String, ?> attributes) {
         // Valid for primative types, but needs to change when a value is an object or an array
         Object userAttributeValue = attributes.get(name);
 
-        if (value != null) { // if there is a value in the condition
-            // check user attribute value is equal
-            return value.equals(userAttributeValue);
+        if (!"custom_attribute".equals(type)) {
```

Comment:
Once we support other condition types we'll probably want to do this check in ProjectConfig or wherever. But this is fine for now. I'm not overly concerned about the fact that we're constructing CustomAttribute objects for things that might not semantically represent `custom_attribute` conditions.